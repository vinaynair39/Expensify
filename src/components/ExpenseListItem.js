import React from 'react';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import moment from 'moment';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>{numeral(amount).format('0,0')} - {moment(createdAt).format("Do MMM YYYY")}</p>
  </div>
);


export default ExpenseListItem;
