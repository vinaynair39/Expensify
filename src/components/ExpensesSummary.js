import React from 'react';
import getVisibleExpenses from '../selectors/expenses';
import getTotalAmount from '../selectors/expenses-total';
import {connect} from 'react-redux';
import numeral from 'numeral';


export const ExpensesSummary = (props) => {
    const word = props.expenses.length <= 1 ?  'expense' : 'expenses';
    return(
        <h3>Viewing {props.expenses.length} {word} totaling: {numeral(getTotalAmount(props.expenses)).format('0,0')}</h3>
    );
}

const mapStateToProps = (state) => {
    return{
        expenses: getVisibleExpenses(state.expenses, state.filters),
    }
}

export default connect(mapStateToProps)(ExpensesSummary);