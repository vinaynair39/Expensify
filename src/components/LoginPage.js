import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export const LoginPage = (props) => {
    return(
        <div>
            <h1>
                <button onClick = {props.Login}>Login</button>
                <button>Sign Up!</button>
            </h1>
        </div>
    );
};

const mapDispatchToProps = (dispatch) => {
    return {
        Login : () => dispatch(startLogin()),
    };
};

export default connect(undefined, mapDispatchToProps)(LoginPage);