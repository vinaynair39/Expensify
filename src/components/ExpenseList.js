import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';
import getTotalAmount from '../selectors/expenses-total';

export const ExpenseList = (props) => (
  <div>
    {props.expenses.length === 0 ?   (<p>no expense</p>) : props.expenses.map((expense) => {
      return <ExpenseListItem key={expense.id} {...expense} />;
    })}
  
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
