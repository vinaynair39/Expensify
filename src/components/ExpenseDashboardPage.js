import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';
import {NavLink} from 'react-router-dom';

const ExpenseDashboardPage = () => (
  <div>
    <NavLink to="/dashboard" activeClassName="is-active" exact={true}>Dashboard</NavLink>
    <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
    <ExpenseListFilters />
    <ExpensesSummary/>
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
