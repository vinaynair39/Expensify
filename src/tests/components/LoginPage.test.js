import React from 'react';
import {shallow} from 'enzyme';
import {LoginPage} from '../../components/LoginPage';

test('Should render LoginPage properly', () => {
    const wrapper = shallow(<LoginPage/>);
    expect(wrapper).toMatchSnapshot();
});

