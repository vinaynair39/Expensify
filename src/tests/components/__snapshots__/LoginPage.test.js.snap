// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render LoginPage properly 1`] = `
<div>
  <h1>
    <button>
      Login
    </button>
    <button>
      Sign Up!
    </button>
  </h1>
</div>
`;
